tests/functional/builtins/codegen/test_abi_decode_fuzz.py

```diff
@@ -0,0 +1,371 @@
+from dataclasses import asdict, dataclass
+
+import hypothesis as hp
+import hypothesis.strategies as st
+import pytest
+from eth.codecs import abi
+
+from tests.evm_backends.base_env import EvmError
+from vyper.codegen.core import calculate_type_for_external_return, needs_external_call_wrap
+from vyper.semantics.types import (
+    AddressT,
+    BoolT,
+    BytesM_T,
+    BytesT,
+    DArrayT,
+    DecimalT,
+    HashMapT,
+    IntegerT,
+    SArrayT,
+    StringT,
+    TupleT,
+    VyperType,
+    _get_primitive_types,
+    _get_sequence_types,
+)
+from vyper.semantics.types.shortcuts import UINT256_T
+
+from .abi_decode import DecodeError, spec_decode
+
+pytestmark = pytest.mark.fuzzing
+
+type_ctors = []
+for t in _get_primitive_types().values():
+    if t == HashMapT or t == DecimalT():
+        continue
+    if isinstance(t, VyperType):
+        t = t.__class__
+    if t in type_ctors:
+        continue
+    type_ctors.append(t)
+
+complex_static_ctors = [SArrayT, TupleT]
+complex_dynamic_ctors = [DArrayT]
+leaf_ctors = [t for t in type_ctors if t not in _get_sequence_types().values()]
+static_leaf_ctors = [t for t in leaf_ctors if t._is_prim_word]
+dynamic_leaf_ctors = [BytesT, StringT]
+
+MAX_MUTATIONS = 33
+
+
+@st.composite
+# max type nesting
+def vyper_type(draw, nesting=3, skip=None):
+    assert nesting >= 0
+
+    skip = skip or []
+
+    st_leaves = st.one_of(st.sampled_from(dynamic_leaf_ctors), st.sampled_from(static_leaf_ctors))
+    st_complex = st.one_of(
+        st.sampled_from(complex_dynamic_ctors), st.sampled_from(complex_static_ctors)
+    )
+
+    if nesting == 0:
+        st_type = st_leaves
+    else:
+        st_type = st.one_of(st_complex, st_leaves)
+
+    # filter here is a bit of a kludge, would be better to improve sampling
+    t = draw(st_type.filter(lambda t: t not in skip))
+
+    # note: maybe st.deferred is good here, we could define it with
+    # mutual recursion
+    def _go(skip=skip):
+        return draw(vyper_type(nesting=nesting - 1, skip=skip))
+
+    if t in (BytesT, StringT):
+        # arbitrary max_value
+        bound = draw(st.integers(min_value=1, max_value=1024))
```

Comment:
isn't 1024 unnecessarily large so it just slows down the tests? what cases might not be covered by eg 512?